src/vs/workbench/parts/files/common/workingFilesModel.ts

```diff
@@ -300,6 +303,25 @@ export class WorkingFilesModel implements IWorkingFilesModel {
 		return null;
 	}
 
+	public restoreRecentlyRemovedEntry(): WorkingFileEntry {
```

Comment:
I think I would change the data structure of recently closed entries so that it alway is an array (of up to N entries) so that you can perform the reopening multiple times as needed. 

You have to be aware of the fact that you can have untitled files in the list, I suggest to ignore any entry in the working files list of closed entries that is not file thereby.